instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/RabbitChannelInstrumentation.java

```diff
@@ -208,43 +269,28 @@ public static void extractAndStartSpan(
         @Advice.Argument(0) String queue,
         @Advice.Return GetResponse response,
         @Advice.Thrown Throwable throwable,
-        @Advice.Local("otelCallDepth") CallDepth callDepth,
-        @Advice.Local("otelTimer") Timer timer) {
-      if (callDepth.decrementAndGet() > 0) {
-        return;
-      }
-
-      Context parentContext = Java8BytecodeBridge.currentContext();
-      ReceiveRequest request = ReceiveRequest.create(queue, response, channel.getConnection());
-      if (!receiveInstrumenter().shouldStart(parentContext, request)) {
-        return;
-      }
-
-      // can't create span and put into scope in method enter above, because can't add parent after
-      // span creation
-      InstrumenterUtil.startAndEnd(
-          receiveInstrumenter(),
-          parentContext,
-          request,
-          null,
-          throwable,
-          timer.startTime(),
-          timer.now());
+        @Advice.Enter ChannelGetAdviceScope adviceScope) {
+      adviceScope.end(channel, queue, response, throwable);
     }
   }
 
   @SuppressWarnings("unused")
   public static class ChannelConsumeAdvice {
 
+    @Advice.AssignReturned.ToArguments(@Advice.AssignReturned.ToArguments.ToArgument(6))
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void wrapConsumer(
+    public static Object wrapConsumer(
         @Advice.This Channel channel,
         @Advice.Argument(0) String queue,
-        @Advice.Argument(value = 6, readOnly = false) Consumer consumer) {
+        @Advice.Argument(6) Consumer consumer) {
       // We have to save off the queue name here because it isn't available to the consumer later.
-      if (consumer != null && !(consumer instanceof TracedDelegatingConsumer)) {
-        consumer = new TracedDelegatingConsumer(queue, consumer, channel.getConnection());
+      Consumer modifiedConsumer = consumer;
```

Comment:
you don't really need this local, could just return the new value from within the if